modules/frontend/src/main/scala/smithy4s_codegen/components/CodeEditor.scala

```diff
@@ -30,16 +31,21 @@ class CodeEditor() {
                            |  @required
                            |  name: String
                            |}""".stripMargin
-  val codeContent = Var(initial)
+  val codeContent = Var(
+    PermalinkCodec
+      .readOnce()
+      .getOrElse(initial)
+  )
 
   val component =
     div(
       cls := "h-full",
       textArea(
         cls := "block p-2.5 w-full h-full text-sm text-gray-900 bg-gray-50 rounded-lg border border-gray-300 focus:ring-blue-500 focus:border-blue-500 font-mono",
-        value := initial,
+        value <-- codeContent,
         onMountFocus,
-        onInput.mapToValue --> codeContent
+        onInput.mapToValue --> codeContent,
+        PermalinkCodec.read --> codeContent
```

Comment:
thought: so the funny thing is that this works, but IIUC Laminar says it should be using `controller`: https://laminar.dev/documentation#controlled-inputs

I wasn't able to do so, the problem is that there are two sources of values for `codeContent` and they're different types that I couldn't compose.